Dockerfile

```diff
@@ -63,12 +68,31 @@ ENV CGO_ENABLED=0 \
     GOARCH=${TARGETARCH} \
     GOARM=${TARGETVARIANT}
 
-COPY . /go/src/github.com/vmware-tanzu/velero
-
-RUN mkdir -p /output/usr/bin && \
-    export GOARM=$(echo "${GOARM}" | cut -c2-) && \
-    /go/src/github.com/vmware-tanzu/velero/hack/build-restic.sh && \
-    go clean -modcache -cache
+# cache go mod download before applying patches
+RUN --mount=type=cache,target=/go/pkg/mod if [ "${BIN}" = "velero" ]; then \
```

Comment:
This allows second run with a slightly different restic version to reuse some of the prior go build mod cache.